spec/lib/filters/radio_filter_spec.rb

```diff
@@ -43,39 +94,73 @@
   end
 
   describe "#value" do
-    context "when params is present and option_lookup is absent" do
-      let(:params) { %w(alpha) }
-      let(:facet) { {} }
+    context "without option lookup" do
+      context "when an allowed option is provided" do
+        let(:params) { "open_consultations" }
 
-      it "should contain all values" do
-        expect(radio_filter.value).to eq(%w(alpha))
+        it "should return the option as an array" do
+          expect(radio_filter.value).to eq(%w(open_consultations))
+        end
       end
-    end
-
-    context "when params is present and option_lookup is empty" do
-      let(:params) { %w(does_not_exist) }
-      let(:facet) { { "option_lookup" => { "policy_papers" => %w(guidance) } } }
 
-      it "should contain no values" do
-        expect(radio_filter.value).to eq([])
+      context "when no option is provided and a default value is set" do
+        it "should return the default value" do
+          expect(radio_filter.value).to eq(default_value)
+        end
       end
-    end
 
-    context "when params is present and option_lookup is present" do
-      let(:params) { %w(policy_papers) }
-      let(:facet) { { "option_lookup" => { "policy_papers" => %w(guidance) } } }
+      context "when the option is not a string and a default value is set" do
+        let(:params) { [] }
+
+        it "should return the default value" do
+          expect(radio_filter.value).to eq(default_value)
+        end
+      end
 
-      it "should contain all values" do
-        expect(radio_filter.value).to eq(%w(guidance))
+      context "when a disallowed param is provided" do
+        let(:params) { "does_not_exist" }
+
+        context "a default option is set" do
+          it "should return the default value" do
+            expect(radio_filter.value).to eq(default_value)
+          end
+        end
+
+        context "a default option is NOT provided" do
+          let(:allowed_values) { [] }
+          it "should return an empty array" do
+            expect(radio_filter.value).to eq([])
+          end
+        end
       end
     end
 
-    context "when params has multiple values and option_lookup is present" do
-      let(:params) { %w(policy_papers does_not_exist consultations) }
-      let(:facet) { { "option_lookup" => { "consultations" => %w(open closed), "policy_papers" => %w(guidance) } } }
+    context "with option lookup" do
+      let(:facet) {
+        {
+          "option_lookup" => option_lookup,
+          "allowed_values" => allowed_values
+        }
+      }
+
+      context "when no option is selected" do
+        it "should return the corresponding default values from the option_lookup" do
+          expect(radio_filter.value).to eq(%w(guidance))
+        end
+      end
+
+      context "when a disallowed value is provided" do
+        let(:params) { "does_not_exist" }
+        it "should return the corresponding default values from the option_lookup" do
```

Comment:
Is it worth again here splitting out into a default value being provided and not being provided as you have above?

Reply:
Yes good idea. And actually, this unearthed a bug - see the last commit. It was unexpected behaviour for there to be no default value and an option_lookup. Thanks for suggesting this.